app/src/main/java/me/hufman/androidautoidrive/carapp/music/views/PlaybackView.kt

```diff
@@ -366,6 +375,26 @@ class PlaybackView(val state: RHMIState, val controller: MusicController, val ca
 		}
 	}
 
+	private fun redrawRepeatButton() {
+		if (repeatButton != null && controller.isSupportedAction(MusicAction.SET_REPEAT_MODE)) {
+			if (controller.isRepeatingAll()) {
+				repeatButton.getTooltipModel()?.asRaDataModel()?.value = L.MUSIC_TURN_REPEAT_ONE_ON
+				repeatButton.getImageModel()?.asImageIdModel()?.imageId = musicImageIDs.REPEAT_ALL_ON
+			} else if (controller.isRepeatingOne()) {
+				repeatButton.getTooltipModel()?.asRaDataModel()?.value = L.MUSIC_TURN_REPEAT_OFF
+				repeatButton.getImageModel()?.asImageIdModel()?.imageId = musicImageIDs.REPEAT_ONE_ON
+			} else {
+				repeatButton.getTooltipModel()?.asRaDataModel()?.value = L.MUSIC_TURN_REPEAT_ALL_ON
+				repeatButton.getImageModel()?.asImageIdModel()?.imageId = musicImageIDs.REPEAT_OFF
+			}
+			repeatButton.setEnabled(true)
+			repeatButton.setVisible(true)
+		} else if (repeatButton != null) {
+			repeatButton.setEnabled(false)
+			repeatButton.setVisible(false)
```

Comment:
The repeatButton should be left visible and disabled when unavailable, matching the shuffle button, so that the full toolbar of 6 buttons keeps its nice appearance :)

Reply:
Good point, removed the setVisible line!